common/buildcraft/builders/tile/TileQuarry.java

```diff
@@ -358,7 +358,7 @@ public void onPlacedBy(EntityLivingBase placer, ItemStack stack) {
         frameBox.setMin(min);
         frameBox.setMax(max);
         miningBox.reset();
-        miningBox.setMin(new BlockPos(min.getX() + 1, 0, min.getZ() + 1));
+        miningBox.setMin(new BlockPos(min.getX() + 1, CubicChunksChecker.worldMin, min.getZ() + 1));
```

Comment:
What about `min.getY() - 256` or some configurable value instead of the constant?

For vanilla, in like 99.9% of the cases, the below 0 value wouldn't matter here because there is bedrock, and in the tiny percentage of cases someone uses it on skylands (which we all know noone does because issue #3610 existed), it will just go some blocks below y=0, which shouldn't really have any bad effects.

And with cubic chunks, it makes no sense to dig deeper than the terrain is generated, because (at least for now) chunks generated by getBlockState won't have any ores.

Reply:
The problem I noticed when it is -1073741824 on a vanilla world is that after the quarry finishes mining blocks (hits bedrock) some aspects of the game will stop working such as entities freezing, blocks & items cannot be picked up (but blocks can be placed and removed), and the void will not do damage. If the world is reloaded the world will take muck longer to load than usual, locking on 0% for some time until the world loads, and then the effects will go away, but if the quarry is used, such as a new block being placed in the quarry area, and then finishes again, the the same effects will happen.